protostar/starknet/cheatable_cached_state.py

```diff
@@ -0,0 +1,124 @@
+from typing import Dict, List
+
+from starkware.starknet.business_logic.state.state import CachedState
+from starkware.starknet.public.abi import AbiType
+from typing_extensions import Self
+
+from protostar.commands.test.test_environment_exceptions import SimpleReportedException
+from protostar.starknet.types import AddressType, SelectorType, ClassHashType
+
+
+# pylint: disable=too-many-instance-attributes
+class CheatableCachedState(CachedState):
```

Comment:
Was moving `CheatableState` to the `TestExecutionState` problematic, or did you decide not to mix refactoring and adding support for Cairo 0.10?

Reply:
Both 😅